liquibase-standard/src/main/java/liquibase/changelog/StatusChangeLogIterator.java

```diff
@@ -38,6 +41,8 @@ public void run(ChangeSetVisitor visitor, RuntimeEnvironment env) throws Liquiba
         try {
             Scope.child(Scope.Attr.databaseChangeLog, databaseChangeLog, () -> {
                 List<ChangeSet> changeSetList = new ArrayList<>(databaseChangeLog.getChangeSets());
+                TagDatabaseChange tagDatabaseChange = getTagDatabaseChange(changeSetList);
+                boolean thereIsTagDatabaseChange =  tagDatabaseChange != null;
```

Comment:
Those 2 new parameters can be moved inside the `if(Strict)` block below so they don't get computed if they aren't used.

Reply:
Done